File: src/Calendary.Model/Prompt.cs
namespace Calendary.Model;

public class Prompt
{
    public int Id { get; set; }
    public int? ThemeId { get; set; }

    public int? CategoryId { get; set; }

    public string Text { get; set; } = string.Empty;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    // Навігаційні властивості
    public PromptTheme? Theme { get; set; }

    public Category? Category { get; set; }

    public ICollection<PromptSeed> Seeds { get; set; } = [];
}

File: src/Calendary.Ng/src/models/prompt.ts
import { Category } from "./category";
import { PromptSeed } from "./promt-seed";
import { Synthesis } from "./synthesis";

export class Prompt {
    id: number = 0;
    themeId: number = 0;
    categoryId: number = 0;
    text: string = '';
    themeName: string = '';
    seeds: PromptSeed[] = [];
    synthesises: Synthesis[] = [];
    category : Category = new Category();
}